overlord/ifacestate/helpers.go

```diff
@@ -191,24 +191,21 @@ func (c *autoConnectChecker) snapDeclaration(snapID string) (*asserts.SnapDeclar
 }
 
 func (c *autoConnectChecker) check(plug *interfaces.Plug, slot *interfaces.Slot) bool {
-	var plugDecl *asserts.SnapDeclaration
-	if plug.Snap.SnapID != "" {
-		var err error
-		plugDecl, err = c.snapDeclaration(plug.Snap.SnapID)
-		if err != nil {
-			logger.Noticef("error: cannot find snap declaration for %q: %v", plug.Snap.Name(), err)
-			return false
-		}
+	if plug.Snap.SnapID == "" || slot.Snap.SnapID == "" {
```

Comment:
we can't afford to do this unless going forward,  we separate sensible-to-auto-connect vs allow-to-auto-connect